### PR TITLE
feat: filter components from hide-component attribues

### LINKED ISSUE #214: Hide component from header and nav-bar when 'page page-hide-components' is set
We need to hide some component entry from header and from navigation explore menu 

image
image

We could use the  attribute page (based on component configuration file https://github.com/bonitasoft/bonita-documentation-site/issues/689) named `page-hide-components` to do it.

Tasks:

- [x] Create a helper to filter the components array
- [x] Adding a dedicated page on documentation preview site

### PR DESCRIPTION
**Pull request by benjaminParisel**
* Create a new helper to filter components array
* Adding an example page in preview site

closes #214

### CODE REVIEW

File: preview-src/hide-components-entry.adoc [hunk 1]
```diff
@@ -0,0 +1,13 @@
+= Hide components
+:page-hide-components: abc, bonita
+
+
+== Header entry
+
+In the header, the Bonita Theme and Project ABC should be hide on this page. But if you go on another page, the both entry should be available.
```
Comment (rbioteau) [inline preview-src/hide-components-entry.adoc:7]:
```suggestion
In the header, the Bonita Theme and Project ABC should be hidden on this page. But if you go on another page, the both entry should be available.
```

File: preview-src/hide-components-entry.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+
+== Navigation explore
+
+In the navigation explore section (on the bottom-left of the page), the Bonita Theme and Project ABC should be hide on this page. But if you go on another page, the both entry should be available.
```
Comment (rbioteau) [inline preview-src/hide-components-entry.adoc:12]:
```suggestion
In the navigation explore section (on the bottom-left of the page), the Bonita Theme and Project ABC should be hidden on this page. But if you go on another page, the both entry should be available.
```

File: src/partials/header-content.hbs [hunk 3]
```diff
@@ -14,7 +14,7 @@
 
     <div id="topbar-nav" class="navbar-menu">
       {{#unless site.keys.hideNavbarComponentsList}}
-      {{#each site.components}}
+      {{#each (isComponentToHide site.components page) }}
```
Comment (tbouffard) [inline src/partials/header-content.hbs:17]:
```suggestion
      {{#each (isComponentToHide site.components page) }}
```

File: src/partials/header-content.hbs [hunk 4]
```diff
@@ -27,7 +27,7 @@
       <div class="navbar-item search-item hide-for-print">
         <div id="docsearch"></div>
       </div>
-      {{/if}}
+      {{/if}}
```
Comment (tbouffard) [inline src/partials/header-content.hbs:30]:
```suggestion
      {{/if}}
```

**Review by tbouffard — APPROVED** [review]
✔️ Tested with the surge preview environment
✔️ Tested locally in the context of the documentation site (configuration in the component version descriptor): https://github.com/bonitasoft/bonita-documentation-site/pull/691#issuecomment-2033860315

**Pull request merged**